openshift-tools/pkg/discover-etcd-initial-cluster/initial-cluster.go

```diff
@@ -163,13 +166,29 @@ func (o *DiscoverEtcdInitialClusterOptions) Run() error {
 		etcdInitialClusterEntries = append(etcdInitialClusterEntries, fmt.Sprintf("%s=%s", member.Name, member.PeerURLs[0]))
 	}
 	if len(targetMember.Name) == 0 {
+		archiveDataDir(o.DataDir)
 		etcdInitialClusterEntries = append(etcdInitialClusterEntries, fmt.Sprintf("%s=%s", o.TargetName, targetMember.PeerURLs[0]))
 	}
 	fmt.Printf(strings.Join(etcdInitialClusterEntries, ","))
 
 	return nil
 }
 
+func archiveDataDir(dataDir string) error {
+	dir := filepath.Join(dataDir+"-archive", time.Now().Format(time.RFC3339))
+
+	// If dir already exists, add seconds to the dir name
+	if _, err := os.Stat(dir); err == nil {
+		dir = filepath.Join(dataDir+"-archive", time.Now().Add(time.Second).Format(time.RFC3339))
+	}
+	if err := os.Rename(dataDir, dir); err != nil {
```

Comment:
`err != nil && !os.IsNotExist`

Reply:
Oops. Just saw the comments. You are right. For lights-out and upgrade we check early on. 

I am outside now will try to see if I can fix it later.